### PR TITLE
Feat - implement all AudioNode connect / disconnect alternatives

### PR DESCRIPTION
**Pull request by b-ma**
Ready to merge once https://github.com/orottier/web-audio-api-rs/pull/493 is landed

### CODE REVIEW

File: generator/js/AudioNode.tmpl.js [hunk 1]
```diff
@@ -161,21 +198,83 @@ ${d.attributes(d.node).filter(attr => d.name(attr) !== 'context').map(attr => {
...
 
-    if (args[0] instanceof AudioParam) {
-      args[0] = args[0][kNapiObj];
-    }
+    if (arguments.length > 2) {
+      if (args[0] instanceof AudioNode) {
+        const destination = args[0][kNapiObj];
+        const output = conversions['unsigned long'](args[1], {
+          enforceRange: true,
+          context: "Failed to execute 'connect' on 'AudioNode':",
```
Comment (orottier) [inline generator/js/AudioNode.tmpl.js:217]:
```suggestion
          context: "Failed to execute 'disconnect' on 'AudioNode':",
```

File: generator/js/AudioNode.tmpl.js [hunk 2]
```diff
@@ -161,21 +198,83 @@ ${d.attributes(d.node).filter(attr => d.name(attr) !== 'context').map(attr => {
...
+        });
+        const input = conversions['unsigned long'](args[2], {
+          enforceRange: true,
+          context: "Failed to execute 'connect' on 'AudioNode':",
```
Comment (orottier) [inline generator/js/AudioNode.tmpl.js:221]:
```suggestion
          context: "Failed to execute 'disconnect' on 'AudioNode':",
```

File: generator/js/AudioNode.tmpl.js [hunk 3]
```diff
@@ -161,21 +198,83 @@ ${d.attributes(d.node).filter(attr => d.name(attr) !== 'context').map(attr => {
...
+        }
+      } else {
+        throw new TypeError("Failed to execute 'disconnect' on 'AudioNode': : Overload resolution failed");
+      }
+    } else if (arguments.length === 2) {
+      if (args[0] instanceof AudioNode || args[0] instanceof AudioParam) {
+        const destination = args[0][kNapiObj];
+        const output = conversions['unsigned long'](args[1], {
+          enforceRange: true,
+          context: "Failed to execute 'connect' on 'AudioNode':",
```
Comment (orottier) [inline generator/js/AudioNode.tmpl.js:237]:
```suggestion
          context: "Failed to execute 'disconnect' on 'AudioNode':",
```

File: generator/js/AudioNode.tmpl.js [hunk 4]
```diff
@@ -161,21 +198,83 @@ ${d.attributes(d.node).filter(attr => d.name(attr) !== 'context').map(attr => {
...
+
+        try {
+          return this[kNapiObj].disconnect(destination);
+        } catch (err) {
+          throwSanitizedError(err);
+        }
+      } else if (Number.isFinite(args[0])) {
+        const output = conversions['unsigned long'](args[0], {
+          enforceRange: true,
+          context: "Failed to execute 'connect' on 'AudioNode':",
```
Comment (orottier) [inline generator/js/AudioNode.tmpl.js:260]:
```suggestion
          context: "Failed to execute 'disconnect' on 'AudioNode':",
```

File: generator/rs/audio_node.tmpl.rs [hunk 5]
```diff
@@ -137,44 +137,29 @@ macro_rules! audio_node_impl {
...
                 "AudioParam" => {
                     let napi_dest = ctx
                         .env
                         .unwrap::<$crate::audio_param::NapiAudioParam>(&js_dest)?;
                     let native_dest = napi_dest.unwrap();
-                    let _res = native_src.connect_at(native_dest, output as usize, input as usize);
-
-                    Ok(js_dest)
+                    native_src.connect_from_output_to_input(native_dest, output, input);
+                    // proper return value is hanlded on JS side
```
Comment (orottier) [inline generator/rs/audio_node.tmpl.rs:161]:
```suggestion
                    // proper return value is handled on JS side
```

File: generator/rs/audio_node.tmpl.rs [hunk 6]
```diff
@@ -183,18 +168,18 @@ macro_rules! audio_node_impl {
                         &js_dest,
                     )?;
                     let native_dest = napi_dest.unwrap();
-                    let _res = native_src.connect_at(native_dest, output as usize, input as usize);
-
-                    Ok(js_dest)
+                    native_src.connect_from_output_to_input(native_dest, output, input);
+                    // proper return value is hanlded on JS side
```
Comment (orottier) [inline generator/rs/audio_node.tmpl.rs:172]:
```suggestion
                    // proper return value is handled on JS side
```

File: generator/rs/audio_node.tmpl.rs [hunk 7]
```diff
@@ -183,18 +168,18 @@ macro_rules! audio_node_impl {
...
                 ${d.nodes.map(n => { return `"${d.name(n)}" => {
                     let napi_dest = ctx
                         .env
                         .unwrap::<$crate::${d.slug(n)}::${d.napiName(n)}>(&js_dest)?;
                     let native_dest = napi_dest.unwrap();
-                    let _res = native_src.connect_at(native_dest, output as usize, input as usize);
-
-                    Ok(js_dest)
+                    native_src.connect_from_output_to_input(native_dest, output, input);
+                    // proper return value is hanlded on JS side
```
Comment (orottier) [inline generator/rs/audio_node.tmpl.rs:181]:
```suggestion
                    // proper return value is handled on JS side
```

**Review by orottier — APPROVED** [review]
LGTM beside some spelling. Do you have WPT results?

**Comment by b-ma** [comment]
Yup (some of the improvements are due to small fixes I made yesterday)

```
RESULTS:
  - # pass: 6927
  - # fail: 677
  - # type error issues: 4
> wpt duration: 1:50.123 (m:ss.mmm)
```

**Comment by orottier** [comment]
I have just released `v1.0.0-rc.5` (and also as `0.43.0` because crates.io/docs.rs do not show RC versions by default)

**Pull request merged**